### ISSUE TITLE
PyPy and Eventlet

### ISSUE THREAD

**Issue by bjmb** [eventlet/eventlet#402]
Hi,

I'm running into this issue: https://github.com/gevent/gevent/pull/496 (getting `AttributeError: 'Semaphore' object has no attribute '_py3k_acquire'`), but I'm using Python 2.7.13 and PyPy 5.7.0. Also I've noticed the the lines changed in those commits are not in master anymore, so not sure what's happening.
In particular to get the error I'm running:

```
from eventlet import monkey_patch
monkey_patch()
from threading import Event
ev = Event()
ev.wait(timeout=10)
```

Thanks for you help

**Comment by temoto** [comment]
Please try `ev = eventlet.Event()` without monkey patching.

**Comment by bjmb** [comment]
I get `TypeError: wait() got an unexpected keyword argument 'timeout'`  and works as expected without the timeout argument, but the previous case also worked fine without the timeout argument

**Comment by temoto** [comment]
We don't have a timeout in `Event.wait()` yet.
http://eventlet.net/doc/modules/event.html#eventlet.event.Event.wait

**Comment by bjmb** [comment]
OK I see, thanks for the fast response

**Comment by bjmb** [comment]
But I wonder then why the original snippet works with python2, python3 and even with for example PyPy 2.2.1 and Python 2.7.3

**Comment by temoto** [comment]
@bjmb maybe it used stdlib Event instead. I'm sure you can easily trace why it works if you really want.

I'm bad at explaining things, but you may try this patch to get timeout kwarg.

`pip install https://github.com/eventlet/eventlet/archive/fee4cca8695491c828bf14668426fa8238655e0c.zip`

**Comment by bjmb** [comment]
OK I see, thanks!

**Comment by temoto** [comment]
`Event.wait() timeout=None` support was merged in master 257058bbdd43e4288177ea669023fe357c98d9ba and will be included in next release 0.23

**Comment by gordol** [comment]
i believe this is affecting us too... (via pymongo)

including a full traceback to help others find this:

```
  File "/srv/xxxx/.virtualenvs/xxxx-pypy/site-packages/pymongo/mongo_client.py", line 965, in _socket_for_writes
    server = self._get_topology().select_server(writable_server_selector)
  File "/srv/xxxx/.virtualenvs/xxxx-pypy/site-packages/pymongo/topology.py", line 224, in select_server
    address))
  File "/srv/xxxx/.virtualenvs/xxxx-pypy/site-packages/pymongo/topology.py", line 183, in select_servers
    selector, server_timeout, address)
  File "/srv/  File "/srv/xxxx/.virtualenvs/xxxx-pypy/site-packages/pymongo/mongo_client.py", line 965, in _socket_for_writes
xxxx/.virtualenvs/xxxx-pypy/site-packages/pymongo/topology.py", line 208, in _select_servers_loop
    self._condition.wait(common.MIN_HEARTBEAT_INTERVAL)
  File "/usr/lib/pypy/lib-python/2.7/threading.py", line 347, in wait
    gotit = waiter._py3k_acquire(True, timeout)
    server = self._get_topology().select_server(writable_server_selector)
  File "/srv/xxxx/.virtualenvs/xxxx-pypy/site-packages/pymongo/topology.py", line 224, in select_server
    address))
  File "/srv/xxxx/.virtualenvs/xxxx-pypy/site-packages/pymongo/topology.py", line 183, in select_servers
    selector, server_timeout, address)
  File "/srv/xxxx/.virtualenvs/xxxx-pypy/site-packages/pymongo/topology.py", line 208, in _select_servers_loop
    self._condition.wait(common.MIN_HEARTBEAT_INTERVAL)
  File "/usr/lib/pypy/lib-python/2.7/threading.py", line 347, in wait
    gotit = waiter._py3k_acquire(True, timeout)
```

any ideas as to when 0.23 will be released? looks like there's quite a backlog here: https://github.com/eventlet/eventlet/milestone/8

perhaps an interim 0.23.1 release is in order?

**Comment by temoto** [comment]
@gordol don't worry about backlog, releases happen randomly to sooth people who suffer from their companies bureaucracy. Then all incomplete issues are moved to next milestone, that's how it grew so big.

If you can modify virtualenv requirements, just replace like this:

```diff
-eventlet
+-e git+https://github.com/eventlet/eventlet.git@257058bbdd43e4288177ea669023fe357c98d9ba#egg=eventlet
```

P.S.: 0.23 is in order, 0.23.1 may come after that.

**Comment by gordol** [comment]
> releases happen randomly to sooth people who suffer from their companies bureaucracy. Then all incomplete issues are moved to next milestone, that's how it grew so big.

i'm not sure who that is directed at, and i'm not sure if you are serious, or that is sarcasm...

i meant 0.22.1, obviously. ;)

either way, pinning requirements to a git commit hash is not a foreign concept, but thanks anyway.

**Comment by temoto** [comment]
@gordol no sarcasm, my point is: *granted every commit works* (which is the case in our master branch), every commit is effectively a next release. So there is no point in batching changes to pypi release. Which only now gave me idea that we should automate pypi releases with every master commit, thank you very much.

**Comment by temoto** [comment]
Fix was released on PyPI as 0.22.1
